datalad/distribution/install.py

```diff
@@ -322,21 +322,48 @@ def __call__(dataset=None, path=None, source=None, recursive=False,
 
         # resolve the target location against the provided dataset
         if path is not None:
-            path = resolve_path(path, ds)
+            # make sure it is not a URL, `resolve_path` cannot handle that
+            if is_url(path):
+                try:
+                    path = get_local_path_from_url(path)
+                    path = resolve_path(path, ds)
+                except ValueError:
+                    # URL doesn't point to a local something
+                    pass
+            else:
```

Comment:
There are test repos on github. Could use with_testrepos asking for a network one.

On May 22, 2016 3:10:45 AM EDT, Michael Hanke <notifications@github.com> wrote:
>> @@ -322,21 +322,48 @@ def __call__(dataset=None, path=None,
>source=None, recursive=False,
>>  
>>          # resolve the target location against the provided dataset
>>          if path is not None:
>> -            path = resolve_path(path, ds)
>> +            # make sure it is not a URL, `resolve_path` cannot
>handle that
>> +            if is_url(path):
>> +                try:
>> +                    path = get_local_path_from_url(path)
>> +                    path = resolve_path(path, ds)
>> +                except ValueError:
>> +                    # URL doesn't point to a local something
>> +                    pass
>> +            else:
>
>There is no new behavior within the scope of what we are testing. To
>test this code path above I'd need to have something served from a
>non-local URL (explained in the commit message). I tried to figure out
>how to use `serve_path_via_http()` which might be able to accomplish
>that ,but failed to infer its function from the tests (need to serve a
>particular path), and docs are not available.
>
>---
>You are receiving this because you commented.
>Reply to this email directly or view it on GitHub:
>https://github.com/datalad/datalad/pull/443/files/4a0cba415a77889b1561fa6220d1f88020c7b694#r64145231